t/plugin/proxy-cache.t

```diff
@@ -0,0 +1,630 @@
+qr/404 Not Found/
+--- response_headers
+Apisix-Cache-Status: MISS
+--- no_error_log
+[error]
+
+
+
+=== TEST 16: hit route (404 there's no cache indeed)
+--- request
+GET /hello-not-found
+--- error_code: 404
+--- response_body eval
+qr/404 Not Found/
+--- response_headers
+Apisix-Cache-Status: MISS
+--- no_error_log
+[error]
+
+
+
+=== TEST 17: hit route (HEAD method)
+--- request
+HEAD /hello-world
+--- error_code: 200
+--- response_headers
+Apisix-Cache-Status: MISS
+--- no_error_log
+[error]
+
+
+
+=== TEST 18: hit route (HEAD method there's no cache)
+--- request
+HEAD /hello-world
+--- error_code: 200
+--- response_headers
+Apisix-Cache-Status: MISS
+--- no_error_log
+[error]
+
+
+
+=== TEST 19:  hide cache headers = false
+--- config
+       location /t {
+           content_by_lua_block {
+               local t = require("lib.test_admin").test
+               local code, body = t('/apisix/admin/routes/1',
+                    ngx.HTTP_PUT,
+                    [[{
+                        "plugins": {
+                            "proxy-cache": {
+                               "cache_zone": "disk_cache_one",
+                               "cache_bypass": ["$arg_bypass"],
+                               "cache_method": ["GET"],
+                               "cache_http_status": [200],
+                               "hide_cache_headers": false,
+                               "no_cache": ["$arg_no_cache"]
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1986": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello*"
+                   }]]
+                   )
+
+               if code >= 300 then
+                   ngx.status = code
+               end
+               ngx.say(body)
+           }
+       }
+--- request
+GET /t
+--- error_code: 200
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 20: hit route (catch the cache headers)
+--- request
+GET /hello
+--- response_body chop
+hello world!
+--- response_headers
+Apisix-Cache-Status: HIT
+--- response_headers_like
+Cache-Control:
+--- no_error_log
+[error]
+
+
+
+=== TEST 21: purge cache
+--- request
+PURGE /hello
+--- error_code: 200
+--- no_error_log
+[error]
+
+
+
+=== TEST 22: purge cache (not found)
+--- request
+PURGE /hello-world
+--- error_code: 404
+--- no_error_log
+[error]
+
+
+
+=== TEST 23:  invalid cache zone
+--- config
+       location /t {
+           content_by_lua_block {
+               local t = require("lib.test_admin").test
+               local code, body = t('/apisix/admin/routes/1',
+                    ngx.HTTP_PUT,
+                    [[{
+                        "plugins": {
+                            "proxy-cache": {
+                               "cache_zone": "invalid_disk_cache",
```

Comment:
We should throw an error message directly when the user tries to set an invalid `cache zone`.

Reply:
I've also considered whether a message should be thrown.  But the check_schema function cannot get variable.  Currently return 500 is the Nginx default behavior.

Reply:
Any suggestions?

Reply:
We can scan the `cache_zone` data from the local config file `conf/config.yaml`.

We just check if the name is valid.

https://github.com/apache/incubator-apisix/pull/1153/files#diff-82e3bcd74a1f6e15d0577acef2d4fa8cR41

Reply:
@membphis  OK. Does the parsing way refer to the bin/apisix file?

Reply:
We can get the local configuration this way. Here is an example:

https://github.com/apache/incubator-apisix/blob/aa3c71a10ad9c37dcb86653a6a4ef325fa86097b/lua/apisix/plugins/heartbeat.lua#L84